src/components/appointment/AppointmentPlanBasicForm.tsx

```diff
@@ -14,30 +14,68 @@ const messages = defineMessages({
   hoursAgo: { id: 'appointment.label.hoursAgo', defaultMessage: '小時前' },
   daysAgo: { id: 'appointment.label.daysAgo', defaultMessage: '天前' },
   appointmentDeadline: {
-    id: 'appointment.text.appointmentDeadline.',
+    id: 'appointment.text.appointmentDeadline',
     defaultMessage: '限定用戶於時段開始前多久需完成預約',
   },
+  isReschedule: {
+    id: 'appointment.text.isReschedule',
+    defaultMessage: '是否允許更換預約時段',
+  },
 })
 
 type FieldProps = {
   title: string
   phone: string
   reservationAmount: number
   reservationType: ReservationType
+  rescheduleAmount: number
+  rescheduleType: ReservationType
   meetGenerationMethod: MeetGenerationMethod
 }
 
+const UpdateAppointmentPlan = gql`
+  mutation UpdateAppointmentPlan(
+    $appointmentPlanId: uuid!
+    $title: String!
+    $phone: String!
+    $reservationAmount: numeric
+    $reservationType: String
+    $rescheduleAmount: Int
+    $rescheduleType: String
+    $meetGenerationMethod: String
+  ) {
+    update_appointment_plan(
+      where: { id: { _eq: $appointmentPlanId } }
+      _set: {
+        title: $title
+        phone: $phone
+        reservation_amount: $reservationAmount
+        reservation_type: $reservationType
+        reschedule_amount: $rescheduleAmount
+        reschedule_type: $rescheduleType
+        meet_generation_method: $meetGenerationMethod
+      }
+    ) {
+      affected_rows
+    }
+  }
+`
+
 const AppointmentPlanBasicForm: React.FC<{
   appointmentPlanAdmin: AppointmentPlanAdminProps | null
   onRefetch?: () => void
 }> = ({ appointmentPlanAdmin, onRefetch }) => {
   const { formatMessage } = useIntl()
   const [form] = useForm<FieldProps>()
+  const [isReschedule, setIsReschedule] = useState<boolean>(false)
 
-  const [updateAppointmentPlanTitle] = useMutation<
-    hasura.UPDATE_APPOINTMENT_PLAN_TITLE,
-    hasura.UPDATE_APPOINTMENT_PLAN_TITLEVariables
-  >(UPDATE_APPOINTMENT_PLAN_TITLE)
+  useEffect(() => {
+    setIsReschedule(appointmentPlanAdmin?.rescheduleAmount !== -1)
+  }, [appointmentPlanAdmin?.rescheduleAmount])
```

Comment:
I think useEffect is not necessary, the default state can be `useState(appointmentPlanAdmin?.rescheduleAmount !== -1)`

Reply:
OK, adjusted.